api-1.10/src/main/java/org/openmrs/module/reporting/cohort/definition/evaluator/DrugOrderCohortDefinitionEvaluator.java

```diff
@@ -0,0 +1,113 @@
+import org.openmrs.annotation.Handler;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import java.util.List;
+import java.util.Date;
+
+@Handler(supports = { DrugOrderCohortDefinition.class })
+public class DrugOrderCohortDefinitionEvaluator implements CohortDefinitionEvaluator {
+
+	@Autowired
+	EvaluationService evaluationService;
+
+	public DrugOrderCohortDefinitionEvaluator() {
+	}
+
+	public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context) {
+		DrugOrderCohortDefinition drugOrderCohortDefinition = (DrugOrderCohortDefinition) cohortDefinition;
+		context = ObjectUtil.nvl(context, new EvaluationContext());
+
+		HqlQueryBuilder query = new HqlQueryBuilder();
+		query.select("drugOrder.patient.patientId");
+		query.from(DrugOrder.class, "drugOrder");
+
+		query.wherePatientIn("drugOrder.patient.patientId", context);
+	
+	    if (drugOrderCohortDefinition.getOnlyCurrentlyActive() != null) {
+	    	if (drugOrderCohortDefinition.getOnlyCurrentlyActive() == true) {
+	    		query.whereNotNull("drugOrder.dateActivated").and().whereGreaterOrNull("drugOrder.dateStopped", new Date());  
+	    	}
+	    	else if (drugOrderCohortDefinition.getOnlyCurrentlyActive() == false) {
+		    	  query.whereNotNull("drugOrder.dateActivated").and().whereLess("drugOrder.dateStopped", new Date());
+	    	}
+	    }
+	    if (drugOrderCohortDefinition.getDrugSets() != null) {
+	    	
+	    	if (drugOrderCohortDefinition.getWhich() == "ANY") {
+		    	query.whereInAny("drugOrder.concept", drugOrderCohortDefinition.getDrugSets().toArray());
+	    	} 
+	    	else if (drugOrderCohortDefinition.getWhich() == "ALL") {
+		    	query.whereIn("drugOrder.concept", drugOrderCohortDefinition.getDrugSets());
+		    	query.groupBy(
+			        		"drugOrder.patient.patientId" + " having count(distinct drugOrder.concept.conceptId) = " + drugOrderCohortDefinition.getDrugSets().size());
+	    	} 
+	    	else if (drugOrderCohortDefinition.getWhich() == "NONE") {
+		    	query.whereNotInAny("drugOrder.concept", drugOrderCohortDefinition.getDrugSets());
+	    	}
+	    }
+	
+	    if (drugOrderCohortDefinition.getDrugConcepts() != null) {
+	    	if (drugOrderCohortDefinition.getWhich().equals("ANY")) {
+	    		query.whereInAny("drugOrder.concept", drugOrderCohortDefinition.getDrugConcepts().toArray());
+	    	} 
+	    	else if (drugOrderCohortDefinition.getWhich().equals("ALL")) {
+	    		query.whereIn("drugOrder.concept", drugOrderCohortDefinition.getDrugConcepts());
+	    		query.groupBy(
+	    				"drugOrder.patient.patientId" + " having count(distinct drugOrder.concept.conceptId) = " + drugOrderCohortDefinition.getDrugSets().size());
+	    	} 
+	    	else if (drugOrderCohortDefinition.getWhich() == "NONE") {
+	    		query.whereNotInAny("drugOrder.concept", drugOrderCohortDefinition.getDrugConcepts());
+	    	}
+	    }
+	
+	    if (drugOrderCohortDefinition.getActivatedOnOrBefore() != null) {
```

Comment:
@mseaton , thank you  for the review. On this note, I am thinking that these null checks are definitely needed since not all the parameters may be specified before the query is fired and that some may remain null (and these need to be skipped in the query building).

Reply:
The null checks may be addressed by the `HqlQueryBuilder` class but meanwhile comparisons within the `DrugOrderCohortDefinitionEvaluator` class can't be made with null references if not all the parameters are defined before building the query.